src/providers/secondary-media-loader.ts

```diff
@@ -0,0 +1,59 @@
+import ILoader = KalturaPlayerTypes.ILoader;
+
+const {RequestBuilder} = KalturaPlayer.providers;
+
+export class SecondaryMediaLoader implements ILoader {
+  _parentEntryId: string = '';
+  _requests: any[] = [];
+  _response: any = {};
+
+  static get id(): string {
+    return 'dualscreen';
+  }
+
+  /**
+   * @constructor
+   * @param {Object} params loader params
+   */
+  constructor(params: any) {
+    this._parentEntryId = params.parentEntryId;
+    const headers: Map<string, string> = new Map();
+    const request = new RequestBuilder(headers);
+    const INCLUDE_FIELDS = 1;
+    request.service = 'baseEntry';
+    request.action = 'list';
+    request.params = {
+      filter: {objectType: 'KalturaBaseEntryFilter', parentEntryIdEqual: this._parentEntryId},
+      responseProfile: {
+        type: INCLUDE_FIELDS,
+        fields: 'id'
+      }
+    };
+    this.requests.push(request);
+  }
+
+  set requests(requests: any[]) {
+    this._requests = requests;
+  }
+
+  get requests(): any[] {
+    return this._requests;
+  }
+
+  set response(response: any) {
+    this._response.entryId = response[0].data.objects[0].id;
```

Comment:
It's possible to do that but it will not change much the code.
This line will look instead like this:
`this._response.entryId = new KalturaBaseEntryListResponse(response[0].data).entries[0].id;`
Also KalturaMediaEntry inside it is a real overkill and expects to parse a lot of properties and I filter only Id in the response. Also if tomorrow we need another field to a plugin I will need to add it to the provider object and release core for this.

Reply:
@RoyBregman I do not understand why you say it will be an overkill.
This is exactly done to provide typed objects and avoid breakage of deeply nested objects.
It will not look like 
`this._response.entryId = new KalturaBaseEntryListResponse(response[0].data).entries[0].id;`
If we keep saying that we will need to override the base objects each time we will start writing more and more again over and over again.
The base entry object is quite lean and includes most of the fields of the base entry object so will cover most cases.
The parsing logic and error checking logic in the core library will make sure we do not get to places where we access `undefined` fields.

The way it has been till now was to include the full blown client libs that include all of this and more and now we are switching to kaltura player providers that include a thin layer which is better then nothing.

Reply:
Why do think the code will not look like this after using providers response types?
Look here for example:
https://github.com/kaltura/playkit-js-providers/blob/master/src/k-provider/ovp/loaders/media-entry-loader.js#L48
Also KalturaBaseEntryListResponse wraps the errors internally in a manner that will not bubble to the dual screen.
Look here:
https://github.com/kaltura/playkit-js-providers/blob/master/src/k-provider/ovp/response-types/kaltura-base-entry-list-response.js#L23
I do see the point of reusing ServiceResult class though as like you said it handles the error checking.
In addition, KalturaBaseEntry just copies the fields one by one w/o any additional logic / validations. so except flow typing (which doesn't help me when using TypeScript) I don't see much benefit.